src/bindings/c/include/openvino/c/ov_model.h

```diff
@@ -0,0 +1,148 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief This is a header file for ov_model C API, which is a C wrapper for ov::Model class.
+ * A user-defined model.
+ * @file ov_model.h
+ */
+
+#pragma once
+
+#include "openvino/c/ov_common.h"
+#include "openvino/c/ov_node.h"
+#include "openvino/c/ov_partial_shape.h"
+
+typedef struct ov_model ov_model_t;
+
+// Model
+/**
+ * @defgroup model model
+ * @ingroup openvino_c
+ * Set of functions representing of Model and Node.
+ * @{
+ */
+
+/**
+ * @brief Release the memory allocated by ov_model_t.
+ * @ingroup model
+ * @param model A pointer to the ov_model_t to free memory.
+ */
+OPENVINO_C_API(void) ov_model_free(ov_model_t* model);
+
+/**
+ * @brief Get the outputs of ov_model_t.
+ * @ingroup model
+ * @param model A pointer to the ov_model_t.
+ * @param output_nodes A pointer to the ov_output_nodes.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e) ov_model_outputs(const ov_model_t* model, ov_output_node_list_t* output_nodes);
+
+/**
+ * @brief Get the outputs of ov_model_t.
+ * @ingroup model
+ * @param model A pointer to the ov_model_t.
+ * @param input_nodes A pointer to the ov_input_nodes.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e) ov_model_inputs(const ov_model_t* model, ov_output_node_list_t* input_nodes);
+
+/**
+ * @brief Get the outputs of ov_model_t.
+ * @ingroup model
+ * @param model A pointer to the ov_model_t.
+ * @param tensor_name input tensor name (char *).
+ * @param input_node A pointer to the ov_output_const_node_t.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_model_input_by_name(const ov_model_t* model, const char* tensor_name, ov_output_const_node_t** input_node);
+
+/**
+ * @brief Get the outputs of ov_model_t.
+ * @ingroup model
+ * @param model A pointer to the ov_model_t.
+ * @param index input tensor index.
+ * @param input_node A pointer to the ov_input_node_t.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_model_input_by_index(const ov_model_t* model, const size_t index, ov_output_const_node_t** input_node);
+
+/**
+ * @brief Returns true if any of the op's defined in the model contains partial shape.
+ * @param model A pointer to the ov_model_t.
+ */
+OPENVINO_C_API(bool) ov_model_is_dynamic(const ov_model_t* model);
+
+/**
+ * @brief Do reshape in model with a list of <name, partial shape>.
+ * @ingroup model
+ * @param model A pointer to the ov_model_t.
+ * @param tensor_names input tensor name (char *) list.
+ * @param partialShape A PartialShape list.
+ * @param cnt The item count in the list.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_model_reshape(const ov_model_t* model,
+                 const char* tensor_names[],
+                 const ov_partial_shape_t* partial_shapes[],
```

Comment:
why arrays as passed by value?

Reply:
It is array pointer.

Reply:
Is there any difference? Ok, let's align them.